packages/mobile/src/home/WalletHome.tsx

```diff
@@ -1,212 +1,156 @@
-import { Balances, balancesSelector } from 'src/stableToken/selectors'
+import { balancesSelector } from 'src/stableToken/selectors'
 import { FeedType } from 'src/transactions/TransactionFeed'
 import TransactionsList from 'src/transactions/TransactionsList'
 import { Currency, STABLE_CURRENCIES } from 'src/utils/currencies'
 import { checkContactsPermission } from 'src/utils/permissions'
 
-interface StateProps {
-  loading: boolean
-  recipientCache: NumberToRecipient
-  numberVerified: boolean
-  cashInButtonExpEnabled: boolean
-  balances: Balances
-  showTokensInHome: boolean
-}
-
-interface DispatchProps {
-  refreshAllBalances: typeof refreshAllBalances
-  initializeSentryUserContext: typeof initializeSentryUserContext
-  setLoading: typeof setLoading
-  showMessage: typeof showMessage
-  importContacts: typeof importContacts
-}
-
-type Props = StateProps & DispatchProps & WithTranslation
-
-const mapDispatchToProps = {
-  refreshAllBalances,
-  initializeSentryUserContext,
-  setLoading,
-  showMessage,
-  importContacts,
-}
-
-const mapStateToProps = (state: RootState): StateProps => ({
-  loading: state.home.loading,
-  recipientCache: phoneRecipientCacheSelector(state),
-  numberVerified: state.app.numberVerified,
-  cashInButtonExpEnabled: state.app.cashInButtonExpEnabled,
-  balances: balancesSelector(state),
-  showTokensInHome: multiTokenShowHomeBalancesSelector(state),
-})
-
 const AnimatedSectionList = Animated.createAnimatedComponent(SectionList)
 
-interface State {
-  isMigrating: boolean
-}
-
-export class WalletHome extends React.Component<Props, State> {
-  scrollPosition: Animated.Value<number>
-  onScroll: (event: NativeSyntheticEvent<NativeScrollEvent>) => void
-
-  constructor(props: Props) {
-    super(props)
+function WalletHome() {
+  const { t } = useTranslation(Namespaces.walletFlow5)
+
+  const appState = useSelector(appStateSelector)
+  const isLoading = useSelector((state) => state.home.loading)
+  const recipientCache = useSelector(phoneRecipientCacheSelector)
+  const showTokensInHome = useSelector(multiTokenShowHomeBalancesSelector)
+  const balances = useSelector(balancesSelector)
+  const cashInButtonExpEnabled = useSelector((state) => state.app.cashInButtonExpEnabled)
+
+  const scrollPosition = useRef(new Animated.Value(0)).current
+  const onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: scrollPosition } } }])
+
+  const dispatch = useDispatch()
+
+  const showTestnetBanner = () => {
+    dispatch(
+      showMessage(
+        t('testnetAlert.1', { testnet: _.startCase(DEFAULT_TESTNET) }),
+        ALERT_BANNER_DURATION,
+        null,
+        null,
+        t('testnetAlert.0', { testnet: _.startCase(DEFAULT_TESTNET) })
+      )
+    )
+  }
 
-    this.scrollPosition = new Animated.Value(0)
-    this.onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: this.scrollPosition } } }])
-    this.state = {
-      isMigrating: false,
+  const tryImportContacts = async () => {
```

Comment:
do we no longer want to check if the user's number is verified before importing contacts?

Reply:
Yeah, I don't know why I changed that, didn't mean to, undid that change, thanks!